### PR TITLE
XCM Precomplies - doc update

### PR DESCRIPTION
**Pull request by PierreOssun**
XCM precompiles v2 doc update with examples

### CODE REVIEW

File: docs/build/EVM/precompiles/xcm.md [hunk 1]
```diff
@@ -1,9 +1,220 @@
...
+
+Find it [here](https://github.com/AstarNetwork/EVM-XCM-Examples/tree/main).
+This repository contains examples demonstrating solidity contracts using XCM precompiles. It's an easy way to start if you want to understand and build with EVM & XCM
+Inside the repository:
+- Zombienet config file: spawn a local zombienet with one relay chain and two parcahins (Shibuya and Shiden node)
+- A setup script in order to create an asset and register it in both networks
+- Solidity examples of usage of XCM precompiles
+- Integration tests (hardaht) in order to understand the flow of the examples
+
+Please follow the instruction ins the README to try it on your local machine.
```
Comment (gitofdeepanshu) [inline docs/build/EVM/precompiles/xcm.md:81]:
```suggestion
Please follow the instruction in the README to try it on your local machine.
```

File: docs/build/EVM/precompiles/xcm.md [hunk 2]
```diff
@@ -1,9 +1,220 @@
...
+XCM.Multilocation memory beneficiary = XCM.Multilocation({
+    parents: 0,
+    interior: interior
+});
+```
+
+#### 3. asset address
+This is the precompile address of asset id = 1 (address = '0xFFFFFFFF...' + DecimalToHex(AssetId)): `0xFfFFFFff00000000000000000000000000000001`. The precompiles expect an array of assets.
+```solidity
+ddress[] memory assetId = new address[](1);
```
Comment (gitofdeepanshu) [inline docs/build/EVM/precompiles/xcm.md:191]:
```suggestion
address[] memory assetId = new address[](1);
```

**Review by gitofdeepanshu — APPROVED** [review]
LGTM
nice docs, just some minor typos to be fixed.

**Review by gluneau — APPROVED** [review]
Added changes of my own as well

**Pull request merged**